### PR TITLE
fix: UNIFY-1138, UNIFY-1130 loading unconfigured project

### PR DESCRIPTION
**Pull request by tgriesser**
- Closes UNIFY-1138
- Closes UNIFY-1130

The `ProjectLifecycleManager` should be self-contained in managing the config loading based on state changes, we shouldn't be calling `initializeConfig` it `initializeOpenMode` as we might not have a configured project.

### CODE REVIEW

**Comment by brian-mann** [comment]
<3 those beautiful tests

File: packages/launchpad/cypress/e2e/open-mode.cy.ts [hunk 1]
```diff
@@ -18,6 +18,34 @@ describe('Launchpad: Open Mode', () => {
...
+    cy.visitLaunchpad()
+    cy.get('h1').should('contain', 'Project Setup')
+  })
+
+  it('opens correctly in unconfigured project with --component', () => {
+    cy.scaffoldProject('pristine')
+    cy.openProject('pristine', ['--component'])
+    cy.visitLaunchpad()
+    cy.get('h1').should('contain', 'Project Setup')
+  })
```
Comment (emilyrohrbough) [inline packages/launchpad/cypress/e2e/open-mode.cy.ts:33]: I'd expect to find project setup test should live in the ProjectSetup spec and the migration tests in the migration spec file.

Comment (tgriesser) [reply packages/launchpad/cypress/e2e/open-mode.cy.ts:33]: Good point, not really sure why I chose this one, I think I thought I was in a different file

**Review by emilyrohrbough — APPROVED** [review]

**Comment by cowboy** [comment]
I don't have comments about the code, but this was not behaving exactly as I had expected, see my comment here https://cypress-io.atlassian.net/browse/UNIFY-1130?focusedCommentId=16511

**Comment by marktnoonan** [comment]
I haven't pulled this down yet, but for visibility, I think the issue @cowboy is seeing is unrelated, and is covered here: https://cypress-io.atlassian.net/browse/UNIFY-1115

**Review by marktnoonan — APPROVED** [review]
Looks good. I'm going to grab UNIFY-1115 today to update the copy for E2E Project Setup, so it's more clear that everything is working as expected, and then we can add that copy to these tests.

**Comment by brian-mann** [comment]
I just read through the specs - I didn't have a chance to review/approve. My assumption based on the way the tests are written is that passing the testing type on an unconfigured project only takes you to the "Project Setup" page instead of the "Configure Testing Type". 

The logic based on how we wrote the CLI is that the flag acts as a commitment for a specific testing type - it's akin to you "clicking the testing type" in the UI - and therefore I would assume it would take you to "Configure the project".

However - I just looked through the CLI logic here: https://docs.google.com/spreadsheets/d/1e3hgmVTWSN3pz_JSIyxgAgqJsGJq5ausKZYALgzUR3U/edit#gid=0

Per that row - it does indicate that you should be on "Choose your testing type", and now I'm wondering why I decided that. BUT that row also indicates that there should be a warning displayed telling the user the testing type is not configured. This has parity with how `cypress run --component` works (except that does not warn, it errors). I imagine that's been lost as a requirement at some point.

**Pull request merged**